### PR TITLE
CST-387

### PR DESCRIPTION
**Pull request by suluvai**
# Summary

This is on request from Unity to removed unused (not part of interface) endpoints.

Updated openapi spec to not include non-readonly endpoints
 Built and tested example running fine.

# Why the changes

_State the reason/context for the change._

# Things worth calling out

_Give useful tips/gotchas/trade-offs made to the reviewers._

# Before merging

- [ ] **_For Immutable developers:_** Do the changes in this PR require documentation updates? Please refer to this [SDK documentation guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide) and list links to documentation update PRs (they don't have to be merged) below (or write `N/A`):
  - [ ] _Add documentation update 1_
  - [ ] _Add documentation update 2_

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,11 +15,11 @@ Includes the API methods implemented for the following readonly endpoints:
...
-  - Trades: Get & List
-  - Orders: Get & List
-  - Users: Get
+  - NFT Checkout Primary: Get & List transactions, Get currencies -
+  - Deposits: Get & List - Done
+  - Withdrawals: Get & List - Done
+  - Transfers: Get & List - Done
+  - Trades: Get & List - Done
+  - Orders: Get & List - Done
+  - Users: Get - Done
```
Comment (suluvai) [inline CHANGELOG.md]: Will fix this `done`

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,11 +15,11 @@ Includes the API methods implemented for the following readonly endpoints:
...
   - Mints: Get & List
-  - NFT Checkout Primary: Get & List transactions, Get currencies
-  - Deposits: Get & List
-  - Withdrawals: Get & List
-  - Transfers: Get & List
-  - Trades: Get & List
-  - Orders: Get & List
-  - Users: Get
+  - NFT Checkout Primary: Get & List transactions, Get currencies -
+  - Deposits: Get & List - Done
```
Comment (Nathan-Ballantyne) [inline CHANGELOG.md:19]: Should the "Done" comments be removed?

File: Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj [hunk 3]
```diff
@@ -29,7 +25,6 @@
     <PackageReference Include="RestSharp" Version="108.0.2" />
     <PackageReference Include="Polly" Version="7.2.3" />
     <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
-    <None Include="..\..\..\..\..\README.md" Pack="true" PackagePath="\"/>
```
Comment (Nathan-Ballantyne) [inline Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj:32]: These will need to added back in?

File: Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj [hunk 4]
```diff
@@ -6,21 +6,17 @@
     <AssemblyName>Imx.Sdk.Gen</AssemblyName>
     <PackageId>Imx.Sdk.Gen</PackageId>
     <OutputType>Library</OutputType>
-    <Authors>Immutable</Authors>
-    <Company>Immutable</Company>
-    <AssemblyTitle>Immutable C# Core SDK API Library</AssemblyTitle>
```
Comment (CodeSchwert) [inline Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj:11]: @suluvai do these need to be updated manually?

File: Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj [hunk 5]
```diff
@@ -29,7 +25,6 @@
     <PackageReference Include="RestSharp" Version="108.0.2" />
     <PackageReference Include="Polly" Version="7.2.3" />
     <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
-    <None Include="..\..\..\..\..\README.md" Pack="true" PackagePath="\"/>
-    <None Include="..\..\..\..\..\LICENSE" Pack="true" PackagePath="\"/>
```
Comment (Nathan-Ballantyne) [inline Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj:33]: These will need to back added back in?
`    <None Include="..\..\..\..\..\README.md" Pack="true" PackagePath="\"/>
    <None Include="..\..\..\..\..\LICENSE" Pack="true" PackagePath="\"/>
`

File: Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj [hunk 6]
```diff
@@ -6,21 +6,17 @@
     <AssemblyName>Imx.Sdk.Gen</AssemblyName>
     <PackageId>Imx.Sdk.Gen</PackageId>
     <OutputType>Library</OutputType>
-    <Authors>Immutable</Authors>
-    <Company>Immutable</Company>
-    <AssemblyTitle>Immutable C# Core SDK API Library</AssemblyTitle>
-    <Description>A library generated from a OpenAPI</Description>
+    <Authors>OpenAPI</Authors>
+    <Company>OpenAPI</Company>
+    <AssemblyTitle>OpenAPI Library</AssemblyTitle>
```
Comment (Nathan-Ballantyne) [inline Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj:11]: Change to Immutable?

File: Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj [hunk 7]
```diff
@@ -6,21 +6,17 @@
...
+    <Description>A library generated from a OpenAPI doc</Description>
```
Comment (carleeto) [inline Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj]:
```suggestion
    <Description>A library generated from an OpenAPI doc</Description>
```

File: Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj [hunk 8]
```diff
@@ -6,21 +6,17 @@
...
-    <Version>0.1.0</Version>
+    <Version>0.1.1</Version>
     <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\Imx.Sdk.Gen.xml</DocumentationFile>
     <RepositoryUrl>https://github.com/immutable/imx-core-sdk-csharp.git</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
     <PackageReleaseNotes>Minor update</PackageReleaseNotes>
-    <IsPackable>true</IsPackable>
-    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-    <PackageReadmeFile>README.md</PackageReadmeFile>
-    <PackageLicenseFile>LICENSE</PackageLicenseFile>
```
Comment (carleeto) [inline Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj:23]: Package README.md and LICENSE required?

File: Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj (hunk 4 above)
Comment (suluvai) [reply Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj]: We can potentially look at updating the template file to include these automatically.

Comment (suluvai) [reply Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj]: Will create a task

Comment (suluvai) [reply Src/IMX/Api/src/Imx.Sdk.Gen/Imx.Sdk.Gen.csproj]: https://immutable.atlassian.net/browse/DX-1729

**Review by CodeSchwert — APPROVED** [review]

**Review by carleeto — APPROVED** [review]

**Review by Nathan-Ballantyne — APPROVED** [review]

**Pull request merged**